File: exemplos_aula07/c/ex1.c
#include <stdio.h>
#include <pthread.h>

int global;
void *thr_func(void *arg);

int main(void){
	pthread_t tid;

	global = 20;

	printf("Thread principal: %d\n", global);
	pthread_create(&tid, NULL, thr_func, NULL);
	pthread_join(tid, NULL);
	printf("Thread principal: %d\n", global);

	return 0;
}

void *thr_func(void *arg){
	global = 40;
	printf("Nova thread: %d\n", global);

	return NULL;
}

File: exemplos_aula07/java/Ex1.java
import java.io.*;

public class Ex1 {
	static volatile int global = 20;

	static class Ex1Thread implements Runnable {
		public void run(){
			global = 40;

			System.out.println("Nova thread: " + global);
		}
	}

	public static void main(String[] args) {
		System.out.println("Thread principal: " + global);
		Thread t = new Thread(new Ex1Thread());
		t.start();
		try {
			t.join();
		} catch(Exception ex) {
		}
		System.out.println("Thread principal: " + global);

	}
}
